### PR TITLE
drivers/shtcx: converted the shtc1 driver into shtcx and added shtc3 support

### PR DESCRIPTION
**Pull request by viktorbatista**
### Contribution description

Pretty self explanatory: conversion of the shtc1 module into the shtcx, with the shtc1 and shtc3 drivers integrated as pseudomodules.
While the SHTC3 sensor is backwards compatible with SHTC1 software, it features a few extra features not yet included, mostly related to a new low power mode. 
The Kconfig file was also updated to suit the new module.

### Testing procedure

Just flashing the board with the test software and getting the output onto a terminal. Then simply blowing on it to see if it is working. Below is the console output with debug on:
Capture

### Issues/PRs references

-

### CODE REVIEW

**Comment by benpicco** [comment]
I would have preferred two commits, one for the renaming and one for adding support for shtc3 - it's a bit hard to tell what the changes are this way.

**Comment by benpicco** [comment]
Please give this a rebase, there is a merge conflict now

**Review by benpicco — DISMISSED** [review]
Looks good to me

**Comment by benpicco** [comment]
CI is not happy :(

File: dist/tools/doccheck/exclude_patterns [hunk 1]
```diff
@@ -10031,7 +10031,7 @@ drivers/include/sht2x\.h:[0-9]+: warning: Member SHT2X_USER_EOB_MASK \(macro def
 drivers/include/sht2x\.h:[0-9]+: warning: Member SHT2X_USER_HEATER_MASK \(macro definition\) of group drivers_sht2x is not documented\.
 drivers/include/sht2x\.h:[0-9]+: warning: Member SHT2X_USER_OTP_MASK \(macro definition\) of group drivers_sht2x is not documented\.
 drivers/include/sht2x\.h:[0-9]+: warning: Member SHT2X_USER_RESERVED_MASK \(macro definition\) of group drivers_sht2x is not documented\.
-drivers/include/shtc1\.h:[0-9]+: warning: unbalanced grouping commands
+drivers/include/shtcx\.h:[0-9]+: warning: unbalanced grouping commands
```
Comment (gschorcht) [inline dist/tools/doccheck/exclude_patterns:10034]: Instead of adding this warning to the exclusion patterns, it should be fixed. Unlike the "Warning: Member ...", this can cause problems with the document structure.

**Review by gschorcht — CHANGES_REQUESTED** [review]

**Comment by gschorcht** [comment]
This driver only supports clock-stretching, as the SHTC1 driver did before, which I also don't like for the following reasons:
- Clock stretching is not supported by all I2C peripherals.
- It blocks the I2C bus for 10 ms in Normal Mode.
- It blocks lower priority threads for that time if a I2C bitbanging software implementation is used.

Therefore I would recommend to use the chance that the driver is touched anyway, to change it to start a in an I2C transaction the measurement, set a timer and then read out the measurement data after a status check in a second I2C transaction.

**Comment by benpicco** [comment]
> Therefore I would recommend to use the chance that the driver is touched anyway, to change it to start a in an I2C transaction the measurement, set a timer and then read out the measurement data after a status check in a second I2C transaction.

I would rather have that done in a separate PR as it’s unrelated to shtc3 support.
It’s hard to follow the changes with the renaming already - let’s keep things simple.

**Comment by gschorcht** [comment]
> I would rather have that done in a separate PR as it’s unrelated to shtc3 support.
> It’s hard to follow the changes with the renaming already - let’s keep things simple.

Of course, it was just a recommendation. I just have the concern that as soon as this PR is merged, no one will have that on the agenda anymore.

The reason why request changes is the documentation problem that has to be fixed https://github.com/RIOT-OS/RIOT/pull/17699#discussion_r823344763.

File: drivers/include/shtcx.h [hunk 2]
```diff
@@ -99,10 +101,10 @@ int8_t shtc1_read(const shtc1_t *dev, uint16_t *rel_humidity, int16_t *temperatu
  * @param[in] dev       The I2C device descriptor.
  * @param[in] id        ID of the device.
  *
- * @return              SHTC1_OK on everything done.
- * @return              SHTC1_ERROR_BUS on error.
+ * @return              SHTCX_OK on everything done.
```
Comment (gschorcht) [inline drivers/include/shtcx.h:104]: Please add a line break in `@details, the line is longer than 100 characters. Feel free to squash it directly.

**Comment by benpicco** [comment]
There is still a [problem with Kconfig](https://ci.riot-os.org/RIOT-OS/RIOT/17699/0e5652c2cfc8c67c834fd46f05d40555ca7eb543/output/compile/tests/driver_ccs811/samr21-xpro:gnu.txt)

File: drivers/shtcx/Kconfig [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+config HAVE_SHTC1
+    bool
+    select HAVE_SHTCX
+    help
+        Indicates that an SHTC1 sensor is present.
+
+config HAVE_SHTC3
+    bool
+    select HAVE_SHTC3
```
Comment (leandrolanzieri) [inline drivers/shtcx/Kconfig:57]:
```suggestion
    select HAVE_SHTCX
```

File: dist/tools/doccheck/exclude_patterns (hunk 1 above)
Comment (viktorbatista) [reply dist/tools/doccheck/exclude_patterns:10034]: Done
